### PR TITLE
resolving coffeecup issues

### PR DESCRIPTION
**Pull request by ayyoob**

### CODE REVIEW

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js [hunk 1]
```diff
@@ -17,244 +17,302 @@
...
+        getData();
+    } else {
+        var backendApiUrl = $("#connectedcup-div-chart").data("backend-api-url") + "/sensors/temperature" + "?from=" + from + "&to=" + to;
+        var successCallback = function (data) {
+            if (data) {
+                drawTemperatureLineGraph(JSON.parse(data));
             }
+        };
+        invokerUtil.get(backendApiUrl, successCallback, function (message) {
+            console.log(message);
```
Comment (prabathabey) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:234]: Replace console logging as appropriate. Stick to jaggery log module for any logging related need.

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js [hunk 2]
```diff
@@ -17,244 +17,302 @@
...
 
+    function drawCoffeeLevelLineGraph(data) {
+        if (data.length === 0 || data.length === undefined) {
+            return;
+        }
+
+        var chartData = [];
+        for (var i = 0; i < data.length; i++) {
+            chartData.push(
+                {
```
Comment (prabathabey) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:308]: Fix formatting issues in this particular line.

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js (hunk 1 above)
Comment (ayyoob) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:234]: fixed with https://github.com/wso2/product-iots/pull/181/commits/9c82589af2a0d7c28a3e85aba7c7417747b370a7

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js (hunk 2 above)
Comment (ayyoob) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:308]: Fixed with https://github.com/wso2/product-iots/pull/181/commits/9c82589af2a0d7c28a3e85aba7c7417747b370a7

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js [hunk 3]
```diff
@@ -17,244 +17,302 @@
  */
 
 var palette = new Rickshaw.Color.Palette({scheme: "classic9"});
-var graphMap = {};
 
 function drawGraph_connectedcup(from, to) {
+    $("#y_axis-temperature").html("");
```
Comment (dakshika) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:22]: .empty() is much faster than html() empty. refer this jsperf 

https://jsperf.com/jquery-html-vs-empty-vs-innerhtml/9

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js [hunk 4]
```diff
@@ -17,244 +17,302 @@
...
-        streamIndex++;
+    } else {
+        temperatureGraphConfig['series'].push(
+                {
+                    'color': palette.color(),
+                    'data': [{
+                        x: parseInt(new Date().getTime() / 1000),
+                        y: 0
+                    }],
+                    'name': $("#connectedcup-details").data("devicename")
```
Comment (dakshika) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:99]: it's better to cache $("#connectedcup-details")

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js (hunk 3 above)
Comment (ayyoob) [inline modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js]: fixed with https://github.com/wso2/product-iots/pull/190/commits/418ed9baae8a91bc0384ef6f1fcd18205b969f3d

Comment (ayyoob) [reply modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:22]: fixed it with https://github.com/wso2/product-iots/pull/190/commits/b8335a0a38f1f77f64dd8cb8d84f6cb4930c5011

File: modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js (hunk 4 above)
Comment (ayyoob) [reply modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js:99]: fixed it with https://github.com/wso2/product-iots/pull/190/commits/418ed9baae8a91bc0384ef6f1fcd18205b969f3d

**Pull request merged**